### ISSUE TITLE
static byte arrays not decoded properly

### ISSUE THREAD

**Issue by rlaveycal** [eris-ltd/eris-contracts.js#21]
If I have a contract

```
    function getBytes() constant returns (byte[10]){
        byte[10] memory b;
        string memory s = "hello";
        bytes memory sb = bytes(s);

        uint k = 0;
        for (uint i = 0; i < sb.length; i++) b[k++] = sb[i];
        b[9] = 0xff;
        return b;
    }
```

then when I use it I get

```
 [ '68', '00', '00', '00', '00', '00', '00', '00', '00', '00' ]
```

rather than

```
 [ '68', '65', '6C', '6C', '6F', '00', '00', '00', '00', 'FF' ]
```

The problem seems to lie in `type.js` and the `decode()` method, specifically the part that splits the byte string into parts and decodes each one

```
            var length = self.staticArrayLength(name);                      // in int
            var arrayStart = offset;                                        // in bytes

            var nestedName = self.nestedName(name);
            var nestedStaticPartLength = self.staticPartLength(nestedName) ; // in bytes
            var result = [];

            for (var i = 0; i < length * nestedStaticPartLength; i += nestedStaticPartLength) {
                result.push(self.decode(bytes, arrayStart + i, nestedName));
            }
```

`self.staticPartLength(nestedName)` evaluates as 1 causing the loop to increment by 1 byte (2 chars) in the byte string each time rather than by 32 bytes (64 characters). Making this change

```
            var nestedStaticPartLength = self.staticPartLength(nestedName) * 32; // in bytes
```

seems to fix it but may not be the most elegant solution.

This is with eris-contracts 0.13.1

**Comment by rlaveycal** [comment]
There's actually a deeper problem - try a contact that returns ``` (byte[12][100] s, uint count) ```. The bytes will decode correctly but the integer won't as the offset isn't calculated properly from the byte array (it calculates it as 12 rather than 12 * 100 * 32)

**Comment by NodeGuy** [comment]
I've reproduced the bugs and suspect they may be in upstream again.  I'm investigating.

**Comment by NodeGuy** [comment]
The first issue is fixed in the `beta` branch.

The second issue is still a bug.

**Comment by NodeGuy** [comment]
The second bug is in web3.  I've created [an issue](https://github.com/ethereum/web3.js/issues/503) for them.  I'll leave this one open until they fix it (or we fix it for them).

**Comment by NodeGuy** [comment]
This will be fixed in 0.16.
